### PR TITLE
Add useFill parameter to CreateScreenshot

### PR DESCRIPTION
**Pull request by alexchuber**
CreateScreenshot currently letterboxes the rendering canvas in the screenshot. The `useFill` property, if true, will fill the given `size` dimension of the screenshot with the rendering canvas. If the rendering canvas' aspect ratio does not match the aspect ratio of the dimensions, then it will be clipped to fit. 

Other:
- Early return if using `CreateScreenshotUsingRenderTarget`
- Update comments

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/16013/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16013/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16013/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/16013/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/16013/merge
https://gui.babylonjs.com/?snapshot=refs/pull/16013/merge
https://nme.babylonjs.com/?snapshot=refs/pull/16013/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/16013/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16013/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/16013/merge/testResults/webgl2playwright/index.html

File: packages/dev/core/src/Misc/screenshotTools.ts [hunk 1]
```diff
@@ -40,7 +41,8 @@ export function CreateScreenshot(
     successCallback?: (data: string) => void,
     mimeType = "image/png",
     forceDownload = false,
-    quality?: number
+    quality?: number,
+    useFill = false
```
Comment (ryantrem) [inline packages/dev/core/src/Misc/screenshotTools.ts:45]: Would it make sense to introduce an overload that takes an options object?

Comment (alexchuber) [reply packages/dev/core/src/Misc/screenshotTools.ts:45]: Makes sense to me. But maybe in a separate PR?

File: packages/dev/core/src/Misc/screenshotTools.ts [hunk 2]
```diff
@@ -95,32 +77,57 @@ export function CreateScreenshot(
...
+            Logger.Error("Failed to create screenshot. Rendering context or rendering canvas is not available.");
+            return;
+        }
+
+        const srcWidth = renderingCanvas.width;
+        const srcHeight = renderingCanvas.height;
+        const destWidth = screenshotCanvas.width;
+        const destHeight = screenshotCanvas.height;
+
+        // Calculate scale factors for width and height. Use the larger of the two to fill the screenshot.
```
Comment (Popov72) [inline packages/dev/core/src/Misc/screenshotTools.ts:102]: I think the second part of the comment is not accurate, as it depends on the `useFill` parameter value?

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Misc/screenshotTools.ts (hunk 2 above)
Comment (alexchuber) [reply packages/dev/core/src/Misc/screenshotTools.ts]: Oops, you're right. Updated.

**Pull request merged**